### ISSUE TITLE
Moving theorems for basic linear algebra from mathboxes into main part 

### ISSUE THREAD

**Issue by avekens** [metamath/set.mm#1333]
(This issue was originally #1314, but was copied because the discussion in #1314 has gone into a differnt direction)

I almost finished the revisions of the sections with theorems in the context of the Cayley Hamilton theorem in my mathbox. The next step would be to move these sections (and after the definition of diagonal/scalar matrices is clarified, see the discussion in the Google group https://groups.google.com/d/msg/metamath/4_ZOT4Gddwo/k-lxGMxGBgAJ , also subsection "The subalgebras of diagonal and scalar matrices " ) into the main part, Part "11. BASIC LINEAR ALGEBRA").

I would propose to place subsection "The subalgebras of diagonal and scalar matrices" after subsection "11.2.1 The matrix algebra", and sections "Polynomial matrices" and "The characteristic polynomial " after "11.3 The determinant".

Would this be OK? Of course there will be a lot of auxiliary theorems currently in my mathbox, which are used in the proofs of these sections, to be moved to the main part, too.

**Comment by avekens** [comment]
Post to #1314 on 17-Dec-2019, from @avekens:
According to David's Google group post https://groups.google.com/d/msg/metamath/jINlwEjvwOc/IINkUI1sBAAJ, I want to move the Cayley Hamilton theorem and all the theorems required to prove it from my mathbox into the main part of set.mm, as indicated in this issue. Before I can do this, the question how to define diagonal and scalar matrices must be answered, see my Google group post https://groups.google.com/d/msg/metamath/4_ZOT4Gddwo/k-lxGMxGBgAJ . Unfortunately, there is no response to this post until now. So I want to ask (some of) you directly:

@nmegill @benjub @tirix @david-a-wheeler @digama0 : do you have any hint/opinion/preference which of the alternatives to define diagonal and scalar matrices described in the Google group post https://groups.google.com/d/msg/metamath/4_ZOT4Gddwo/k-lxGMxGBgAJ should be used?

As soon as we have a decision, I'll start to revise section "The subalgebras of diagonal and scalar matrices" in my mathbox. Afterwards, I can move this and other subsections concerning the Cayley Hamilton theorem into the main part as proposed in this issue.

**Comment by avekens** [comment]
Post to #1314 on 17-Dec-2019, from @tirix:

> Both solutions described in your post seem to have their advantages and drawbacks.
> 
> I think my vote may go to your current ‘ DMatALT ‘, since its definition is simpler, the operations are identical, and I would assume that later developments would use more often statements like “M is a diagonal matrix” rather than “the set of diagonal matrices forms a ring”.
> 
> However, I have no strong opinion and would agree with any solution.

Post to #1314 on 17-Dec-2019, from @benjub:

> I do not have enough experience in long proofs and extensible structures to give an authorized advice. From a theoretical point of view, I would prefer the first version.

**Comment by avekens** [comment]
Post to #1314 on 18-Dec-2019, from @avekens:
Coming back to the original question, Thierry had good arguments for defining substructures as sets (and not structures). Meanwhile, I would also prefer these definitions (DMatALT, ScMatALT), because they better fit to the way substructures are currently used in set.mm. Furthermore, it will be easier to revise the existing theorems about diagonal/scalar matrices, because they are based on the set-like definitions already.

If nobody has arguments for a structure-like definition, I will switch the names of the definitions (ALT vs. non-ALT) and start to revise the theorems about diagonal/scalar matrices using the set-like definitions. @nmegill @digama0 @david-a-wheeler @sorear what do you think?

**Comment by avekens** [comment]
With PR #1332, the issue with the definitions of diagonal and scalar matrices is solved.
The next step will be to move section "The subalgebras of diagonal and scalar matrices" to the main part of set.mm.

**Comment by david-a-wheeler** [comment]
@avekens - I don't have any hint/opinion/preference on which of the alternatives to define diagonal and scalar matrices. I'd look to Norm & Mario.

**Comment by avekens** [comment]
Sections "The subalgebras of diagonal and scalar matrices" and "Matrices of dimension 0 and 1" were moved to the main part of set.mm, see PR #1336.
The next step will be to move section "Polynomial matrices"  to the main part of set.mm.

**Comment by avekens** [comment]
Section "Polynomial matrices" was moved to the main part of set.mm, see PR #1343 .

**Remark:** I have split subsection "The infinite sequence builder "seq"" into subsections "Miscellaneous theorems about integers" and "The infinite sequence builder "seq"", because many of my auxiliary theorems depend on theorems of the first part, which is a collection of very different theorems about integers, not directly related to infinite sequence builders (the definition df-seq was provided in the middle of the former subsection, and I made the cut directly above it). Maybe another one finds a better name for the new subsection "Miscellaneous theorems about integers", or has an idea how this subsection can be restructured.
  
The next step will be to move the last section "The characteristic polynomial" in the context of the Cayley-Hamilton theorem to the main part of set.mm.
